### PR TITLE
French (France) locale

### PR DESCRIPTION
**Pull request by Prometheos2**
I'm open to suggestions, and problems about typos, grammar or text length.

It should also serve as a solid basis (_I hope_) in case it is not a proper Canadian French.

- [x] src/_locales
- [x] assets/texts

### CODE REVIEW

**Comment by Prometheos2** [comment]
I have, to be honest, no clue what *internal* means in a release, nor what *featured* could be welly translated in French

**Comment by rugk** [comment]
> internal

= only some internal changes that developers etc. care about. (i.e. that are not visible)

> featured

= a change that I would to highlight, some special, extraordinary etc.

---

However, some general notes:
1. remember the whole add-on is still in beta, so many text strings will change.
2. also, the whole permissions.md description is not even properly written yet (see https://github.com/rugk/awesome-emoji-picker/issues/19)
3. In case it was not wanted by you: Your git commits contain the full name.

And BTW @nicolas-trouin translated my other add-on into French, so maybe it helps to have a look at https://github.com/rugk/offline-qr-code/pull/181. (many strings are shared between add-ons etc.)

**Comment by Prometheos2** [comment]
> Converting the Contributors file to a Markdown one is a good idea, and I take this as a suggestion. However, next time better discuss this change before. I also want to deploy this in other add-ons. ([rugk/offline-qr-code#193](https://github.com/rugk/offline-qr-code/issues/193), [rugk/mastodon-simplified-federation#27](https://github.com/rugk/mastodon-simplified-federation/issues/27))

Oh, sorry :(

> In case it was not wanted by you: Your git commits contain the full name.

Oh, I didn't notice; thanks for the info

**Comment by Prometheos2** [comment]
By the way, would you prefer to keep the original name or to translate it, @rugk?

**Comment by rugk** [comment]
See https://help.github.com/en/articles/changing-author-info for how to rewrite all your commits, if you want.

> By the way, would you prefer to keep the original name or to translate it, @rugk?

Which name? the one of the add-on? If so, rather keep the original English one. :smiley:

**Comment by Prometheos2** [comment]
Well, I already done it the long way :(

That should be done, then

**Comment by rugk** [comment]
So I've added many strings now and adjusted the AMO texts and similar. Feel free to translate them.

**Comment by Prometheos2** [comment]
I don't think I would be able to do that during the week-end; no problem with that?

**Comment by rugk** [comment]
No problem at all. Just v1.0 will then not be in French, but this is fine IMHO.

I will release v1.0 today.
v1.1 is planned soon™ anyway: The usual bugfixes etc. update after 1.0 release. This translation could go into it, then.
The add-on does not have many users anyway and was marked as experimental before, so there won't be that many French users who need to use the English version. :wink: This will likely take some time. (Note the stats are public, so you can see them on AMO.)

**Comment by Prometheos2** [comment]
That should be good.

File: assets/texts/fr/amoDescription.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Ce module complémentaire (extension) pour Firefox vous donne un bon petit sélecteur d'émoticônes pour en trouver et en sélectionner. Il est copié dans votre presse-papier quand vous cliquez dessus.
+Cette application utilise des émoticônes à jour et vous laisse accèder à des émoticônes dans de différents styles que vous connaissez de différents appareils, sites et applications.
+Il est basé sur le superbe <a
```
Comment (rugk) [inline assets/texts/fr/amoDescription.html]: I guess there are a lot of spaces here/not needed line breaks.

File: assets/texts/fr/amoDescription.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Ce module complémentaire (extension) pour Firefox vous donne un bon petit sélecteur d'émoticônes pour en trouver et en sélectionner. Il est copié dans votre presse-papier quand vous cliquez dessus.
```
Comment (rugk) [inline assets/texts/fr/amoDescription.html]: Also the whole `amoDescription.html` file is outdated, e.g. (It was mostly rewritten.)

File: assets/texts/fr/amoScreenshots.csv [hunk 3]
```diff
@@ -0,0 +1 @@
+imageOne.png; "Première image";
```
Comment (rugk) [inline assets/texts/fr/amoScreenshots.csv]: Also outdated…

File: assets/texts/fr/amoDescription.html (hunk 1 above)
Comment (Prometheos2) [reply assets/texts/fr/amoDescription.html]: Oh, sorry; I thought that HTML ignored line breaks without `<br>` like in Markdown 😅

File: assets/texts/fr/amoDescription.html (hunk 2 above)
Comment (Prometheos2) [reply assets/texts/fr/amoDescription.html]: By the way, shouldn't the "good for first issue" be used to award newcomers for doing a good new issue? 🤨

**Comment by Prometheos2** [comment]
I had a another idea of a translation for the issues.

I never truly knew if professional French dev said "problems", "tasks" or something else; it seems to be "ticket".

I will change it quickly.

**Comment by rugk** [comment]
Hmm, AFAIK it would be fine to say "issue". Dev-language is English-centric anyways. Espeically on GitHub, which has no translation anyway. :smile: 

In German, we have all these words too (tickets, tasks, problems), but I anyway just chose to translate it with "issues" in German, as well – it's called "issues" on GitHub. Same with "Fork me on GitHub" = "Fork' mich auf GitHub" (in German).

**Comment by Prometheos2** [comment]
That should be fine then :)

**Comment by rugk** [comment]
Okay, so merged. :smile: 

The add-on translation itself/changes in the add-on will obviously only go into the next version (v1.1 likely), however I've already added the translation on AMO, so it should be live on https://addons.mozilla.org/fr/firefox/addon/awesome-emoji-picker/. :smiley: 

But arrrgh, some follow-up issues:
* the short add-on description for AMO _must only_ be 250 characters long. You have 20 too much:
  image
  (I've now just cut the last part in 1a8a2e9d, but it would obviously be good to find a better solution, if possible that includes all the text.)
* you missed the changelog. The v0.1 is not needed anymore. If you want to translate it, please translate the v1.0.

**Comment by Prometheos2** [comment]
Oh... Sorry about that.

I dunno if we can re-open a PR, but I will work on the problems anyway

**Pull request merged**